pfcpiface/config.go

```diff
@@ -0,0 +1,189 @@
+	EnableFlowMeasure bool             `json:"measure_flow"`
+	SimInfo           SimModeInfo      `json:"sim"`
+	ConnTimeout       uint32           `json:"conn_timeout"` // TODO(max): unused, remove
+	ReadTimeout       uint32           `json:"read_timeout"` // TODO(max): convert to duration string
+	EnableNotifyBess  bool             `json:"enable_notify_bess"`
+	EnableEndMarker   bool             `json:"enable_end_marker"`
+	NotifySockAddr    string           `json:"notify_sockaddr"`
+	EndMarkerSockAddr string           `json:"endmarker_sockaddr"`
+	LogLevel          string           `json:"log_level"`
+	QciQosConfig      []QciQosConfig   `json:"qci_qos_config"`
+	SliceMeterConfig  SliceMeterConfig `json:"slice_rate_limit_config"`
+	MaxReqRetries     uint8            `json:"max_req_retries"`
+	RespTimeout       string           `json:"resp_timeout"`
+	EnableHBTimer     bool             `json:"enable_hbTimer"`
+	HeartBeatInterval string           `json:"heart_beat_interval"`
+}
+
+// QciQosConfig : Qos configured attributes.
+type QciQosConfig struct {
+	QCI                uint8  `json:"qci"`
+	CBS                uint32 `json:"cbs"`
+	PBS                uint32 `json:"pbs"`
+	EBS                uint32 `json:"ebs"`
+	BurstDurationMs    uint32 `json:"burst_duration_ms"`
+	SchedulingPriority uint32 `json:"priority"`
+}
+
+type SliceMeterConfig struct {
+	N6RateBps    uint64 `json:"n6_bps"`
+	N6BurstBytes uint64 `json:"n6_burst_bytes"`
+	N3RateBps    uint64 `json:"n3_bps"`
+	N3BurstBytes uint64 `json:"n3_burst_bytes"`
+}
+
+// SimModeInfo : Sim mode attributes.
+type SimModeInfo struct {
+	MaxSessions uint32 `json:"max_sessions"`
+	StartUEIP   net.IP `json:"start_ue_ip"`
+	StartENBIP  net.IP `json:"start_enb_ip"`
+	StartAUPFIP net.IP `json:"start_aupf_ip"`
+	N6AppIP     net.IP `json:"n6_app_ip"`
+	N9AppIP     net.IP `json:"n9_app_ip"`
+	StartN3TEID string `json:"start_n3_teid"`
+	StartN9TEID string `json:"start_n9_teid"`
+}
+
+// CPIfaceInfo : CPIface interface settings.
+type CPIfaceInfo struct {
+	Peers           []string `json:"peers"`
+	UseFQDN         bool     `json:"use_fqdn"`
+	NodeID          string   `json:"hostname"`
+	EnableUeIPAlloc bool     `json:"enable_ue_ip_alloc"`
+	UEIPPool        string   `json:"ue_ip_pool"`
+	HTTPPort        string   `json:"http_port"`
+	Dnn             string   `json:"dnn"`
+}
+
+// IfaceType : Gateway interface struct.
+type IfaceType struct {
+	IfName string `json:"ifname"`
+}
+
+// validateConf checks that the given config reaches a baseline of correctness.
+func validateConf(conf Conf) error {
+	validModes := map[string]struct{}{
+		"af_xdp":    {},
+		"af_packet": {},
+		"dpdk":      {},
+		"sim":       {},
+	}
+	if _, ok := validModes[conf.Mode]; !ok {
+		return ErrInvalidArgumentWithReason("conf.Mode", conf.Mode, "invalid mode")
+	}
+
+	if conf.EnableP4rt {
+		_, _, err := net.ParseCIDR(conf.P4rtcIface.AccessIP)
+		if err != nil {
+			return err
+		}
+		_, _, err = net.ParseCIDR(conf.P4rtcIface.UEIP)
```

Comment:
I suggest to rename this parameter. 

```suggestion
		_, _, err = net.ParseCIDR(conf.P4rtcIface.UEIPPool)
```

Reply:
+1

luckily the json name is already good
https://github.com/omec-project/upf-epc/blob/59eb8f2e7d3ebe49545d3889422700f17f456186/conf/upf.json#L151